mtags/src/main/scala/scala/meta/internal/pc/HoverMarkup.scala

```diff
@@ -17,9 +17,7 @@ object HoverMarkup {
       forceExpressionType: Boolean = false
   ): String = {
     val markdown = new StringBuilder()
-    val needsExpressionType =
```

Comment:
This caused the changes in HoverDocSuite, but I don't think we should show expression in that case (it also causes some issues with Scala 3, which are currently hard to deal with). We should most likely fix the types in for `emptyList`.

Reply:
Could we revert this and related changes in Scala2 tests and disable `needExpressionType` fallback only for Scala3?
Otherwise, we will have do it later.

I think the case in `HoverDocSuite` is valid, expression type is needed there. I agree that this logic is a bit weird but it works well. Or maybe we can move this clause to HoverProvider in Scala2 and pass `forceExpressionType = pos.start != po.end || !symbolSignature.endsWith(expressionType)` ?